usaspending_api/agency/v2/views/tas_object_class_list.py

```diff
@@ -0,0 +1,151 @@
+            ),
+            # Filters are consistent with object class by agency, DEV-4923
+            Q(
+                Q(financialaccountsbyprogramactivityobjectclass__obligations_incurred_by_program_object_class_cpe__gt=0)
+                | Q(
+                    financialaccountsbyprogramactivityobjectclass__obligations_incurred_by_program_object_class_cpe__lt=0
+                )
+                | Q(
+                    financialaccountsbyprogramactivityobjectclass__gross_outlay_amount_by_program_object_class_cpe__gt=0
+                )
+                | Q(
+                    financialaccountsbyprogramactivityobjectclass__gross_outlay_amount_by_program_object_class_cpe__lt=0
+                )
+            ),
+        ]
+        if self.filter:
+            filters.append(Q(object_class_name__icontains=self.filter))
+        queryset_results = (
+            ObjectClass.objects.filter(*filters)
+            .annotate(
+                name=F("object_class_name"),
+                obligated_amount=Sum(
+                    "financialaccountsbyprogramactivityobjectclass__obligations_incurred_by_program_object_class_cpe"
+                ),
+                gross_outlay_amount=Sum(
+                    "financialaccountsbyprogramactivityobjectclass__gross_outlay_amount_by_program_object_class_cpe"
+                ),
+            )
+            .order_by(f"{'-' if self.pagination.sort_order == 'desc' else ''}{self.pagination.sort_key}")
+            .values("id", "name", "obligated_amount", "gross_outlay_amount")
+        )
+        return queryset_results
+
+    def get_program_activity_by_object_class_list(self, object_class_id) -> List[dict]:
+        submission_ids = get_latest_submission_ids_for_fiscal_year(self.fiscal_year)
+        filters = [
+            Q(financialaccountsbyprogramactivityobjectclass__submission_id__in=submission_ids),
+            Q(financialaccountsbyprogramactivityobjectclass__object_class__id=object_class_id),
+            Q(
+                financialaccountsbyprogramactivityobjectclass__treasury_account__tas_rendering_label=self.tas_rendering_label
+            ),
+            Q(
+                Q(financialaccountsbyprogramactivityobjectclass__obligations_incurred_by_program_object_class_cpe__gt=0)
+                | Q(
+                    financialaccountsbyprogramactivityobjectclass__obligations_incurred_by_program_object_class_cpe__lt=0
+                )
+                | Q(
+                    financialaccountsbyprogramactivityobjectclass__gross_outlay_amount_by_program_object_class_cpe__gt=0
+                )
+                | Q(
+                    financialaccountsbyprogramactivityobjectclass__gross_outlay_amount_by_program_object_class_cpe__lt=0
+                )
+            ),
+        ]
+        if self.filter:
+            filters.append(Q(program_activity_name__icontains=self.filter))
```

Comment:
I believe this filter should only apply to object class names. No need to also apply it to program activities.

Reply:
Ok will do.